### ISSUE TITLE
[Website] On structure and template

### ISSUE THREAD

**Issue by anibalsolon** [INCF/niQC#1]
Hey,

There are an infinity of static-site generator, as you can see at: https://www.staticgen.com/

From what I understood, what are you looking for is Jekyll: https://jekyllrb.com/ (I am not a big fan of the Ruby env, but we can set up a docker dev env)

It supports Markdown, and it has a huge community and examples. We can setup CircleCI to rebuild the website on each merge on the master branch i.e. automatic deploy process.

Giving the initial setup and structure, the maintenance and content creation should be easy.

Let me know what you think about it!

**Comment by raamana** [comment]
no strong preferences, but docusaurus looks good to me
https://docusaurus.io/en/
https://www.staticgen.com/docusaurus

**Comment by anibalsolon** [comment]
I didn't know this one, it looks good! Node + Yarn makes me happy :smile:

**Comment by pbellec** [comment]
No strong preferences either, I have little experience in site building,
and so far I have worked directly with html/CSS templates.

Never heard of docusaurus, but the examples look great.

**Comment by anibalsolon** [comment]
There you go! https://incf.github.io/niQC/

**Comment by raamana** [comment]
awsome! I will try to fill in Aims, and put in some useful links. Suggest what else could be added.

is there a way to achieve a News feed that is easy to manage, and to subscribe with RSS?

I think we need a dedicated page to tools, wherein we maintain a database that can be filtered by modality, type of QC or data...

**Comment by anibalsolon** [comment]
Good, this is their boilerplate so the pages do not make much sense. I will do some cleaning up and improve the template, this low-definition crocodile creeps me out. Do you guys think that we should stick to something close to INCF website design?

It has a built-in RSS feed! https://incf.github.io/niQC/blog/feed.xml
It is based on markdown files under: https://github.com/INCF/niQC/tree/master/website/blog
I will see if it is possible to rename it to "news", but I am not sure it supports different folder names other than "blog".

Do you think the database is going to be large? We can add it directly to the code, just wondering about loading time.

**Comment by raamana** [comment]
Perhaps @pbellec would have more insight into database/tables on a webpage. Or @r03ert0?

**Comment by raamana** [comment]
Hey guys!! How are you? Did you get a chance to work on this yet after the holidays? It's okay if you won't have time soon, just let me know. Thanks.

**Comment by anibalsolon** [comment]
Hey @raamana , I've been studying the Netlify CMS, that is defined as a headless CMS: instead of using a database, it works directly with Github API/git version control to manage content (posts and uploads e.g. images). Using it, we have control of drafts/reviewing and have a GUI to manage all of it (instead of editing code directly). Pretty cool stuff.

This is the structure that I am testing on:
https://github.com/anibalsolon/website-test/

I just need right now to add a markdown compiler (e.g. Jekyll, Hugo) integrated with a continuous integration system to build the .md files.

However, it is possible to continue the website without it for now. I will move away from Docusaurus, as it is not much flexible about the possible types of content (e.g. news, it only has a blog option).

**Comment by raamana** [comment]
thanks Annibal. I am able to follow you only partially here, but perhaps you can dumb it down for me here?

Should and can we use docusaurus for now, or not?
Are you suggesting another CMS to allow dynamic pages? 

Ultimately, we will need something that's not too onerous to maintain e.g. a reasonably tech-savvy person can add few lines to markdown or ReST and push them to update the website on a not-very-frequent basis. Let me know if I can help (my web dev skills are not expert though).

**Comment by anibalsolon** [comment]
Hi @raamana ,

Sorry, I had a deadline.

We can use docusaurus, but I think we should move on to something more flexible and still easy to maintain.

The CMS was actually to provide a nice interface to edit the markdown files and manage publishing (as in, you can keep drafs). It uses the versioning control provided by Git to manage the content. But I guess for now it is not required.

I plan to focus on this today, I will give updates here.

**Comment by anibalsolon** [comment]
Hi @raamana 

It looks like I've reached a good point which we can get back with feedback iterations about layout and organizing niQC content.

https://incf.github.io/niQC/

All the markdown content can be found at: https://github.com/INCF/niQC/tree/master/content

Please let me know when you can check it out!

**Comment by anibalsolon** [comment]
...and now CircleCI works, so changes on master will be compiled automatically and made available into the website :)

**Comment by raamana** [comment]
Sorry for the delay - I was away on vacation.

website looks great and clean! 

did you get a chance to add tables etc?

**Comment by raamana** [comment]
Also, if it helps, we can set up a short skype call to discuss the details to reduce delays in back and forth?

**Comment by anibalsolon** [comment]
Hi @raamana 

I will include this table!

Do you have any time tomorrow? My Skype name is *anibalsolon*

**Comment by raamana** [comment]
Today would work - how about 3pm EST?

**Comment by anibalsolon** [comment]
@raamana works for me!

**Comment by raamana** [comment]
Sounds good - talk to you shortly.

**Comment by raamana** [comment]
Closing this, as we will be moving the work to small individual issues.
